cypress/integration/jsPromisesPracticeDOM.spec.js

```diff
@@ -15,73 +12,59 @@ const thirdResolvedMsg = 'Third promise was resolved';
 
-  it('should ignore middle button click for the second promise', () => {
-    cy.clickButton({ button: 1 });
-    cy.clickButton({ button: 1 });
-    cy.checkMessageDisplayed(secondResolvedMsg).should('not.exist');
-  });
+      page.notification().should('include.text', firstRejectedMsg);
+      page.notification().should('not.include.text', firstResolvedMsg);
+    });
 
-  it('should reject first promise after 3 seconds of inactivity', () => {
-    // NOTE: waiting for reject
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(3000);
-    cy.checkMessageDisplayed(firstRejectedMsg);
-  });
+    it('should not show reject message if it was already resolved', () => {
+      page.body().click();
 
-  it('should resolve second promise on the right click', () => {
-    cy.clickButton({ button: 2 });
-    cy.checkMessageDisplayed(secondResolvedMsg);
+      cy.tick(3000);
+      page.notification().should('not.include.text', firstRejectedMsg);
+    });
   });
 
-  it('should resolve second promise on the left click', () => {
-    cy.clickButton({ button: 0 });
-    cy.checkMessageDisplayed(secondResolvedMsg);
-  });
+  describe('secondPromise', () => {
+    it('should be resolved after the left click', () => {
+      page.body().click();
+      page.notification().should('include.text', secondResolvedMsg);
+    });
 
-  it('should resolve third promise', () => {
-    cy.clickButton({ button: 2 });
-    cy.clickButton({ button: 0 });
-    cy.checkMessageDisplayed(thirdResolvedMsg);
+    it('should be resolved after the right click', () => {
+      page.body().rightclick();
+      page.notification().should('include.text', secondResolvedMsg);
+    });
   });
 
-  it('should resolve all promises', () => {
-    cy.clickButton({ button: 2 });
-    cy.clickButton({ button: 0 });
-    cy.checkMessageDisplayed(firstResolvedMsg);
-    cy.checkMessageDisplayed(secondResolvedMsg);
-    cy.checkMessageDisplayed(thirdResolvedMsg);
-  });
+  describe('thirdPromise', () => {
+    it('should be resolved after the left and right click', () => {
+      page.body().click();
+      page.body().rightclick();
+      page.notification().should('include.text', thirdResolvedMsg);
+    });
+
+    it('should not be resolved after the left click only', () => {
+      page.body().click();
+      page.notification().should('not.include.text', thirdResolvedMsg);
+    });
 
-  it('should reject 1st promise, resolve 2nd and 3rd promises', () => {
-    // NOTE: waiting for reject
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(3000);
-    cy.clickButton({ button: 0 });
-    cy.clickButton({ button: 2 });
-    cy.checkMessageDisplayed(firstRejectedMsg);
-    cy.checkMessageDisplayed(secondResolvedMsg);
-    cy.checkMessageDisplayed(thirdResolvedMsg);
+    it('should not be resolved after the right click only', () => {
+      page.body().rightclick();
+      page.notification().should('not.include.text', thirdResolvedMsg);
+    });
```

Comment:
It would be great to check after multiple clicks not to allow to use just a counter and run it after a second click

Reply:
Also check that it is not resolved after some time